### PR TITLE
feat(auth): GetUserByProviderID()

### PR DESCRIPTION
**Pull request by rsgowman**

### CODE REVIEW

**Comment by rsgowman** [comment]
Note the diffbase. I'll merge+retarget to dev once the bulk_get stuff lands.

File: auth/user_mgt.go
Comment (hiranya911) [inline auth/user_mgt.go]: GetUserByProviderID

File: auth/user_mgt_test.go
Comment (hiranya911) [inline auth/user_mgt_test.go]: You can possibly combine these last 3 test cases into one table-driven test case.

```
var cases []struct{
  providerID string
  providerUID string
  want string
} {
  {
    "email",
    "user@example.com",
   `{"email": ["user@example.com"]}`,
  },
  // Other cases
}
```

File: auth/user_mgt_test.go
Comment (hiranya911) [inline auth/user_mgt_test.go]: Also test for empty providerID, empty providerUID etc

File: auth/user_mgt.go
Comment (rsgowman) [reply auth/user_mgt.go]: Done.

File: auth/user_mgt_test.go
Comment (rsgowman) [reply auth/user_mgt_test.go]: Done.

File: auth/user_mgt_test.go
Comment (rsgowman) [reply auth/user_mgt_test.go]: See below; I added them to the TestInvalidGetUser() test. (Arguably, that could be 4 or 5 different tests, but they are at least very tightly related.)

File: auth/user_mgt_test.go
Comment (rsgowman) [reply auth/user_mgt_test.go]: Done.

File: .github/scripts/publish_post_check.sh
Comment (hiranya911) [inline .github/scripts/publish_post_check.sh:1]: This file no longer exists in source. Can be dropped from this PR.

File: .github/scripts/publish_preflight_check.sh
Comment (hiranya911) [inline .github/scripts/publish_preflight_check.sh:95]: Please revert.

File: .github/workflows/publish.yml
Comment (hiranya911) [inline .github/workflows/publish.yml:1]: Not needed.

File: .github/workflows/stage.yml
Comment (hiranya911) [inline .github/workflows/stage.yml:1]: Not needed.

File: auth/user_mgt_test.go
Comment (hiranya911) [inline auth/user_mgt_test.go:151]: Might want to also check for `IsUserNotFoundError(err)` here

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Implementation and tests look pretty good. Lets remove the redundant CI config bits, and we should be good to go.

File: .github/scripts/publish_post_check.sh
Comment (rsgowman) [reply .github/scripts/publish_post_check.sh]: Ah, that's what caused this bad merge. (I mean, I should've caught it anyways, but this is making more sense now.)

File: .github/scripts/publish_preflight_check.sh
Comment (rsgowman) [reply .github/scripts/publish_preflight_check.sh]: Done.

File: .github/workflows/publish.yml
Comment (rsgowman) [reply .github/workflows/publish.yml]: Done.

File: .github/workflows/stage.yml
Comment (rsgowman) [reply .github/workflows/stage.yml]: Done.

File: auth/user_mgt_test.go
Comment (rsgowman) [reply auth/user_mgt_test.go]: Done.

**Review by hiranya911 — APPROVED** [review]
Thanks! LGTM 👍

File: auth/user_mgt.go
Comment (egilmorez) [inline auth/user_mgt.go:502]: Unless this is a literal, capped "ID" would probably be preferred.

File: auth/user_mgt.go
Comment (egilmorez) [inline auth/user_mgt.go:508]: Suggest "such as" instead of "e.g."

My sense is that if we want to say "For example," it should come after a hard stop like a period or semicolon after "provider."  Internal style guide: https://developers.google.com/style/abbreviations#dont-use

File: auth/user_mgt.go
Comment (egilmorez) [inline auth/user_mgt.go:512]: Should this be IDPs, with the acronym in caps?

Comment (egilmorez) [reply auth/user_mgt.go:512]: Actually, this may not be a public-facing comment I guess :)

**Review by egilmorez — COMMENTED** [review]
Just some minor nits.  See what you think, and I'm happy to approve in any case.  Thanks!

File: auth/user_mgt.go
Comment (rsgowman) [reply auth/user_mgt.go]: Done.

File: auth/user_mgt.go
Comment (rsgowman) [reply auth/user_mgt.go]: Done.

File: auth/user_mgt.go
Comment (rsgowman) [reply auth/user_mgt.go]: Not a public facing comment; but still a valid point, so done anyways.

**Review by egilmorez — APPROVED** [review]
LG, thanks!

**Pull request merged**
